unstable test: TestFinalizer

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```shell
=== RUN   TestFinalizer
    finalizer_test.go:40: 
        	Error Trace:	/home/jenkins/.tidb/tmp/5b8829669b2a941b840f9eb369e2b54f/sandbox/processwrapper-sandbox/4009/execroot/__main__/bazel-out/k8-fastbuild/bin/util/gctuner/gctuner_test_/gctuner_test.runfiles/__main__/util/gctuner/finalizer_test.go:40
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestFinalizer
--- FAIL: TestFinalizer (0.01s)
```
link https://ci.pingcap.net/blue/organizations/jenkins/tidb-unit-test-hourly/detail/tidb-unit-test-hourly/9286/tests
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/util/gctuner/finalizer_test.go
+++ b/util/gctuner/finalizer_test.go
@@ -19,6 +19,7 @@ import (
 	"runtime/debug"
 	"sync/atomic"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/require"
 )
@@ -29,7 +30,7 @@ type testState struct {
 
 func TestFinalizer(t *testing.T) {
 	debug.SetGCPercent(1000)
-	maxCount := int32(16)
+	maxCount := int32(8)
 	state := &testState{}
 	var stopped atomic.Bool
 	defer stopped.Store(true)
@@ -41,6 +42,7 @@ func TestFinalizer(t *testing.T) {
 	})
 	for i := int32(1); i <= maxCount; i++ {
 		runtime.GC()
+		time.Sleep(10 * time.Millisecond)
 		require.Equal(t, i, atomic.LoadInt32(&state.count))
 	}
 	require.Nil(t, f.ref)

